addons/sale_project/models/product.py

```diff
@@ -31,12 +31,8 @@ def _selection_service_policy(self):
         From those, you can track the service you are selling.\n \
         'In sale order\'s project': Will use the sale order\'s configured project if defined or fallback to \
         creating a new project based on the selected template.")
-    project_id = fields.Many2one(
-        'project.project', 'Project', company_dependent=True,
-        domain="[('company_id', '=', current_company_id)]")
-    project_template_id = fields.Many2one(
-        'project.project', 'Project Template', company_dependent=True, copy=True,
-        domain="[('company_id', '=', current_company_id)]")
+    project_id = fields.Many2one('project.project', 'Project')
+    project_template_id = fields.Many2one('project.project', 'Project Template', copy=True)
```

Comment:
We can keep the domain, but the company_dependent is causing issues with the 'task global project' setting. It prevents SO with different companies to create a task in the same project.
e.a. So in company A with product P, SO in company B with product P. If the product is set on 'create task in global project', the task created by the SO in company A will be in project A1, and the 
task created by the SO in company B will be in project A2. I haven't found a solution to prevent that behaviour.